test/jdk/tools/jpackage/helpers/jdk/jpackage/test/LauncherAsServiceVerifier.java

```diff
@@ -162,7 +166,9 @@ static void verify(JPackageCommand cmd) {
 
         for (var launcherAsService : List.of(true, false)) {
             partitionedLauncherNames.get(launcherAsService).forEach(launcherName -> {
-                verify(cmd, launcherName, launcherAsService);
+                // Launcher should not be a service with "--launcher-as-service"
+                // and "--mac-app-store" specified.
+                verify(cmd, launcherName, launcherAsService && !cmd.hasArgument("--mac-app-store"));
```

Comment:
This change looks redundant. In case of "--mac-app-store" the list of launcher names that should be installed as services will (should) be empty after the tweak to the `boolean launcherAsService(JPackageCommand cmd, String launcherName)` function.